discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -515,3 +551,20 @@ def wait_for_cluster_to_be_in_installing_status(self):
             cluster_id=self.id,
             statuses=[consts.ClusterStatus.INSTALLING]
         )
+
+    def get_events(self, host_id=''):
+        return self.api_client.get_events(cluster_id=self.id, host_id=host_id)
+
+    def find_event(self, event_to_find, params_list=[], host_id='', fail_on_error=True):
+        events_list = self.get_events(host_id=host_id)
+        event_exist = False
+        logging.info(f"Searching for event: {event_to_find}")
+        for event in events_list:
+            if event_to_find in event['message']:
+                if all(param in event['message'] for param in params_list):
```

Comment:
I think you should verify that the EventTime is after the test start time or between time ranges - if there is a test that should create the same event 5 times - you will decide that the event_exist even if the event appears only once

Reply:
Done, please review.
Added reference_time parameter